src/Avalonia.Controls/Control.cs

```diff
@@ -530,14 +546,28 @@ protected override void OnKeyUp(KeyEventArgs e)
             }
         }
 
+        /// <inheritdoc/>
         protected override void OnPropertyChanged(AvaloniaPropertyChangedEventArgs change)
         {
             base.OnPropertyChanged(change);
 
-            if (change.Property == FlowDirectionProperty)
+            if (change.Property == BoundsProperty)
+            {
+                var oldValue = change.GetOldValue<Rect>();
+                var newValue = change.GetNewValue<Rect>();
+
+                var sizeChangedEventArgs = new SizeChangedEventArgs(
```

Comment:
Do we want a check here to make sure the sizes are actually different since this comes from Bounds?

Reply:
I assumed that bounds changed event would not be raised unless it was a new value. Since it's coming through the property system... there should be a equality check using the default comparer for the type? Although, I haven't gotten into the details here before and the nuances of "effective value changed".

Reply:
Position of Rect could change but not size which still triggers Bounds.Changed and cause SizeChanged to be fired too.

Reply:
Ah, I understand what you are saying now. Yes, that's correct we don't want to raise the event for X/Y Position changes in the Rect. Code is updated, good catch again.